syncstorage-db-common/src/util.rs

```diff
@@ -112,6 +112,29 @@ impl SyncTimestamp {
     pub fn as_rfc3339(self) -> Result<String, SyncstorageDbError> {
         to_rfc3339(self.as_i64())
     }
+
+    /// Convert this SyncTimestamp into a chrono::NaiveDateTime (UTC) for use with Diesel's `Timestamp`.
+    pub fn as_naive_datetime(self) -> Result<NaiveDateTime, SyncstorageDbError> {
+        let millis = self.as_i64();
+        let secs = millis / 1000;
+        let nsecs: u32 = ((millis % 1000) * 1_000_000).try_into().map_err(|e| {
+            SyncstorageDbError::internal(format!(
+                "Invalid timestamp (nanoseconds) {}: {}",
+                millis, e
+            ))
+        })?;
+
+        let ts = Utc.timestamp_opt(secs, nsecs);
```

Comment:
I think that's the older fn @pjenvey mentioned in our meeting?  Maybe he can remind us why we shouldn't use it.